demos/device_shadow_for_aws/shadow_demo_main.c

```diff
@@ -581,191 +595,214 @@ int RunDeviceShadowDemo( bool awsIotMqttMode,
+                                          SHADOW_TOPIC_LENGTH_DELETE( THING_NAME_LENGTH ),
+                                          pcUpdateDocument,
+                                          0U );
 
-        /* Then we publish a desired state to the /update topic. Since we've deleted
-         * the device shadow at the beginning of the demo, this will cause a delta message
-         * to be published, which we have subscribed to.
-         * In many real applications, the desired state is not published by
-         * the device itself. But for the purpose of making this demo self-contained,
-         * we publish one here so that we can receive a delta message later.
-         */
-        if( xDemoStatus == pdPASS )
-        {
-            /* Desired power on state . */
-            LogInfo( ( "Send desired power state with 1." ) );
+            /* Then try to subscribe shadow topics. */
+            if( xDemoStatus == pdPASS )
+            {
+                xDemoStatus = SubscribeToTopic( &xMqttContext,
+                                                SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
+                                                SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
+            }
 
-            ( void ) memset( pcUpdateDocument,
-                             0x00,
-                             sizeof( pcUpdateDocument ) );
+            if( xDemoStatus == pdPASS )
+            {
+                xDemoStatus = SubscribeToTopic( &xMqttContext,
+                                                SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ),
+                                                SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
+            }
 
-            snprintf( pcUpdateDocument,
-                      SHADOW_DESIRED_JSON_LENGTH + 1,
-                      SHADOW_DESIRED_JSON,
-                      ( int ) 1,
-                      ( long unsigned ) ( xTaskGetTickCount() % 1000000 ) );
+            if( xDemoStatus == pdPASS )
+            {
+                xDemoStatus = SubscribeToTopic( &xMqttContext,
+                                                SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ),
+                                                SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
+            }
 
-            xDemoStatus = PublishToTopic( &xMqttContext,
-                                          SHADOW_TOPIC_STRING_UPDATE( THING_NAME ),
-                                          SHADOW_TOPIC_LENGTH_UPDATE( THING_NAME_LENGTH ),
-                                          pcUpdateDocument,
-                                          ( SHADOW_DESIRED_JSON_LENGTH + 1 ) );
-        }
+            /* This demo uses a constant #THING_NAME known at compile time therefore we can use macros to
+             * assemble shadow topic strings.
+             * If the thing name is known at run time, then we could use the API #Shadow_GetTopicString to
```

Comment:
No action required, this is outside of the scope of the PR – just something to think about:

Why are we demonstrating a niche demo-only use case when most developers will need to build this topic at run time?  The next time I'm in here I'd remove most of lines 641-670 and build the topics dynamically.  You can still use the macro for the thing name, just don't use pre-processor concatenation to build the topics; no one building a real app will be able to do this.

Reply:
Agree. Thanks. Similar to the earlier, this change was not in the scope of this PR, but appeared as a change due to indentation changes. I will create an internal task to handle it. It will be a good improvement to this demo.